### ISSUE TITLE
Access/reference to content in WEB-INF?

### ISSUE THREAD

**Issue by stozk** [jasonish/jetty-springmvc-jsp-template#3]
Hi,

I want to encapsulate a webapp in an embedded jetty to create a standalone executable jar.

I have this webapp working with the jetty plugin from eclipse, but I have trouble porting it to an embedded jetty.

I've tried out your template and I added tiles support, which works so far.
Sometimes it doesn't for some reason and instead of the composed webpage the controller will just return the "regular" index.jsp or xxx.jsp.

In this example you can access the static.txt file in the WEB-INF folder.
I've put a picture in the same folder (icon.png) which cannot be accessed.

I've also tried to put in JS/CSS which also doesn't work because I guess that I'm referencing the context paths in a wrong way or maybe some configuration is missing?

I'm not sure what to do, do you have an idea what's causing this?

Best regards,

Daniel

**Comment by jasonish** [comment]
I've never seen anything like that, but then again, my JSP experience doesn't go much beyond this mini github project.  

Can you replicate it in a minimal project and put it up on Github?  I'd take a look then.

**Comment by stozk** [comment]
I have googled around a bit and it seems that integrating JSP support in an embedded jetty is kinda complicated. Most projects with "working" jsp set them up to pre-compile the jsp when building the project, but that makes using dynamic jsps kinda pointless. It seems to be caused by the fact that it's hard (impossible?) to point the embedded jetty at the taglibs and all that during runtime.

I have uploaded it here:

https://github.com/stozk/embedded-jetty-jsp-spring-mvc-test/tree/master/jetty-springmvc-jsp-template-test

Thanks
Daniel

**Comment by jasonish** [comment]
I'm not too sure at the moment.  But a few things I've dealt with in the past is that JSPs work fine when running from Maven.  But they did not work when running from the jars.  However, they DID work when running from a fatjar.

My solution was to either
1) Precompile the JSPs
2) Run from a fat jar.

But I don't think either of those will help you with dynamic JSPs.  

I was able to get JSPs to work at one time without precompiling, and without a fat jar by exploding the taglibs into my own application jar.  But I felt that was too hacky so settled on precompiling.  Now I actually use Spring Boot and have left all this complicated bootstrapping behind.

Hmm.. Looking at your app a little more, I wonder if WEB-INF is getting in the way, and jetty is doing some blocking there?  What I did was..

Moved the WEB-INF/resources directly under webapp, and then changed:
-       registry.addResourceHandler("/WEB-INF/resources/**").addResourceLocations("/WEB-INF/resources/");
-       registry.addResourceHandler("/resources/**").addResourceLocations("/resources/");

And now I get a nicely styled page with icons.
